crates/ruma-client-api/src/sync/sync_events.rs

```diff
@@ -62,3 +73,174 @@ impl DeviceLists {
         self.changed.is_empty() && self.left.is_empty()
     }
 }
+
+/// Possible event formats that may appear in stripped state.
+#[derive(Debug, Clone)]
+#[cfg_attr(not(ruma_unstable_exhaustive_types), non_exhaustive)]
+#[allow(clippy::large_enum_variant)]
+pub enum StrippedState {
+    /// A stripped state event.
+    Stripped(AnyStrippedStateEvent),
+
+    /// A timeline state event.
+    #[cfg(feature = "unstable-msc4311")]
+    Timeline(AnyStateEvent),
```

Comment:
`Timeline` doesn't seem appropriate for this variant, we use it only to group `MessageLike` and `State` events. For the event format the prefix would be `Stripped`, `Sync` or nothing (and the latter applies here).

Maybe `Full`? We have `FullStateEventContent` which is a bit different but I don't think that it matters (and it is probably not the right name anyway). Or `Event`? It won't conflict with anything, but every format has `Event` in its name.